executor/ddl.go

```diff
@@ -229,13 +230,19 @@ func (e *DDLExec) executeRenameTable(s *ast.RenameTableStmt) error {
 	var err error
 	if len(s.TableToTables) == 1 {
 		oldIdent := ast.Ident{Schema: s.TableToTables[0].OldTable.Schema, Name: s.TableToTables[0].OldTable.Name}
+		if _, ok := e.getLocalTemporaryTable(oldIdent.Schema, oldIdent.Name); ok {
```

Comment:
Maybe you can check that in preprocess or builder or at some ealier stage?
I'm not sure where the global temporary table check this, but it's better for them to be together if possible.

Reply:
For ddl operations, I want it‘s behavior to be as close as possible to mysql. When the ddl constraint check fails, mysql still commits the current transaction (except for create/drop a local temporary table).

Reply:
> When the ddl constraint check fails, mysql still commits the current transaction.
I didn't see you have tested this.